File: app-kotlin/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/firebase/firestore/transactions/user/FirestoreAccount.java
package com.tritiumstudios.phasmophobiaevidencetool.firebase.firestore.transactions.user;

import androidx.annotation.NonNull;

import com.tritiumstudios.phasmophobiaevidencetool.firebase.firestore.transactions.user.account.FirestoreAccountCredit;
import com.tritiumstudios.phasmophobiaevidencetool.firebase.firestore.transactions.user.account.FirestoreTransactionHistory;
import com.google.firebase.firestore.CollectionReference;

public class FirestoreAccount {

    public final static String COLLECTION_ACCOUNT = "Account";

    public static void init()
            throws Exception {
        FirestoreAccountCredit.init();
        FirestoreTransactionHistory.init();
    }

    @NonNull
    public static CollectionReference getAccountCollection()
            throws Exception {
        return FirestoreUser.getUserDocument().collection(COLLECTION_ACCOUNT);
    }

}

File: app-compose/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/core/data/market/palette/source/remote/api/firestore/transactions/user/account/FirestoreAccount.kt
package com.tritiumgaming.phasmophobiaevidencepicker.core.data.market.palette.source.remote.api.firestore.transactions.user.account

import com.google.firebase.firestore.CollectionReference
import com.tritiumgaming.phasmophobiaevidencepicker.core.data.market.palette.source.remote.api.firestore.transactions.user.FirestoreUser.Companion.userDocument
import com.tritiumgaming.phasmophobiaevidencepicker.core.data.market.palette.source.remote.api.firestore.transactions.user.account.properties.FirestoreAccountCredit
import com.tritiumgaming.phasmophobiaevidencepicker.core.data.market.palette.source.remote.api.firestore.transactions.user.account.properties.FirestoreAccountPreferences
import com.tritiumgaming.phasmophobiaevidencepicker.core.data.market.palette.source.remote.api.firestore.transactions.user.account.transactions.FirestoreTransactionHistory

class FirestoreAccount {
    companion object {
        private const val COLLECTION_ACCOUNT: String = "Account"

        val accountCollection: CollectionReference
            @Throws(Exception::class)
            get() = userDocument.collection(COLLECTION_ACCOUNT)

        @Throws(Exception::class)
        fun init() {
            FirestoreAccountCredit.init()
            FirestoreAccountPreferences.init()
            FirestoreTransactionHistory.init()
        }

    }
}
